src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -3,52 +3,45 @@
 import entities.Category;
 import entities.EntityException;
 import entities.MonthlyStorage;
-import views.add_edit_category_views.CategoryP;
 
 import java.util.ArrayList;
 import java.util.NoSuchElementException;
 import java.util.Objects;
 
 /**
- * The Category_Use_Case_Interactor 1. adds/creates a new category and 2. edits an existing category. (Updates MonthlyStorage?)
+ * The Category_Use_Case_Interactor 1. adds/creates a new category and 2. edits an existing category.
+ * Updates MonthlyStorage:
+ * 1. adding a new Category to MonthlyStorage.categoryData with MonthlyStorage.addCategory()
+ * 2.
  * Contains a helper method isValidDouble() and findCategory().
  */
 
 public class CategoryUCI implements CategoryIB {
     final CategoryOB categoryOB;
-    private CategoryOD categoryOD;
 
     /**
      * Constructs CategoryUCI.
      * @param categoryP presenter that is related to the use case.
      */
-
-    public CategoryUCI(CategoryP categoryP) {
+    public CategoryUCI(CategoryOB categoryP) {
         this.categoryOB = categoryP;
-        this.categoryOD = null;
     }
 
     /**
-     * Helper method returns a boolean to check if the input Object is a valid double.
+     * Helper method returns a double to check if the input Object is a valid double.
      * @param value a user input
-     * @return boolean true if value is a valid double, vice versa
+     * @return double converted from value
      */
-
-    public static boolean isValidDouble(Object value){
-        try{
-            Double.parseDouble(String.valueOf(value));
-            return true;
-        }catch (NumberFormatException e){
-            return false;
-        }
+    public static double toDouble(Object value) throws NumberFormatException, NullPointerException{
+            return Double.parseDouble(String.valueOf(value));
     }
     /**
      * Overrides method in Category_IB.
-     * Attempts to add a category with information from CategoryID and returns whether fail/success after execution.
-     * Provides detailed fail messages according to each condition listed below
+     * Attempts to add a category with information from CategoryID and returns a CategoryOD indicating whether fail/success after execution.
+     * Provides detailed fail messages according to each condition listed below:
      *  1. NumberFormatException: User tries to add a new budget value that can not be converted to a double.
      *  2. Category budget less than 0: User tries to add a new budget value that is a negative number.
-     *  3. EntityException: User tries to add an invalid Category but failed. (See entities/EntityException.java)
+     *  3. EntityException: User tries to add an invalid Category name but failed. (See entities/EntityException.java)
```

Comment:
I think she's catching the exception here, so she shouldn't allow it to throw Entity Exception.

Reply:
I thought having the _throws EntityException_ will just throws the exception in both cases, but looking your test case, it seems like it is catching the Exception... Maybe we need to consult our TA.

Reply:
Yeah, I think because you have "throws EntityException", you should mention it in your Javadoc. But maybe we should consult our TA, although I think two possible fixes would be either creating a new exception for one of the two to avoid confusion or getting rid of the EntityException for the monthID (as previously discussed).

Reply:
As discussed in meeting: there are two EntityExceptions. One is thrown before the try catch, and caught in the view. The other is caught in try catch. Thus there shouldn't be a conflict between _throws EntityException_ and _catch (EntityException e)_